database/sql/src/main/scala/cromwell/database/slick/CallCachingSlickDatabase.scala

```diff
@@ -92,12 +92,31 @@ trait CallCachingSlickDatabase extends CallCachingSqlDatabase {
     runTransaction(action)
   }
 
-  override def cacheEntryExistsForCall(workflowExecutionUuid: String, callFqn: String, index: Int)
-                                      (implicit ec: ExecutionContext): Future[Boolean] = {
+  private def callCacheJoinFromEntryQuery(callCachingEntry: CallCachingEntry)
+                            (implicit ec: ExecutionContext) = {
+    val callCachingEntryId = callCachingEntry.callCachingEntryId.get
+    for {
+      callCachingSimpletonEntries <- dataAccess.
+        callCachingSimpletonEntriesForCallCachingEntryId(callCachingEntryId).result
+      callCachingDetritusEntries <- dataAccess.
+        callCachingDetritusEntriesForCallCachingEntryId(callCachingEntryId).result
+      callCachingHashEntries <- dataAccess.
+        callCachingHashEntriesForCallCachingEntryId(callCachingEntryId).result
+      callCachingAggregationEntries <- dataAccess.
+        callCachingAggregationForCacheEntryId(callCachingEntryId).result.headOption
+    } yield CallCachingJoin(callCachingEntry, callCachingHashEntries, callCachingAggregationEntries, callCachingSimpletonEntries, callCachingDetritusEntries)
```

Comment:
ToL could be a (x,y,z) mapN ?

Reply:
Actually I'm not sure if / how this can be done with slick DBIO actions. I'm probably going to need your FP expertise

Reply:
oh the DBIO stuff is not Cats so unless you want to write all the instances... 😬

Reply:
I found this https://github.com/RMSone/slick-cats but it doesn't have an Applicative for DBIO AFAICT so it still doesn't work

Reply:
If you can make a `toErrorOr` for `DBIO` then that it might be easier/less stressful than implementing the typeclass (at least upfront).

If there's a fundamental reason why `DBIO.toErrorOr` wouldn't work, you'd probably run into the same problem trying to make the applicative.

So, I'd probably start with trying to write that conversion method

Reply:
slick-cats' DBIO Monad is all you need for cats to figure out the Applicative ( as it is derived from the `flatMap` in the monad instance)

Reply:
@danbills wait, so why can't we have applicative for `Checked[A]`?

Reply:
@cjllanwarne We most certainly can (and do) have an applicative for it (i.e. `Either`), but what we usually want is the applicative for `Validated`, thus the converting back and forth all the time